compatibilityTests/src/test/scala/org/apache/spark/sql/test/SharedSnappySessionContext.scala

```diff
@@ -27,9 +27,13 @@ import org.apache.spark.sql.test.SharedSnappySessionContext.random
  */
 trait SharedSnappySessionContext extends SharedSQLContext {
 
+  protected def codegenFallback: Boolean = false
+
   override protected def createSparkSession: SnappySession = {
-    val session = new TestSnappySession(sparkConf.set("spark.hadoop.fs.file.impl",
-      classOf[DebugFilesystem].getName).set("snappydata.sql.planCaching.", random.nextBoolean().toString))
+    val session = new TestSnappySession(sparkConf
+        .set("spark.hadoop.fs.file.impl", classOf[DebugFilesystem].getName)
+        .set("spark.sql.codegen.fallback", codegenFallback.toString)
+        .set("snappydata.sql.planCaching.", random.nextBoolean().toString))
```

Comment:
Shouldn't the planCaching be false always?

Reply:
Although by default `planCaching` is disabled, we still support plan caching. This change was done as part of https://github.com/SnappyDataInc/snappydata/pull/1205 to make sure that all test scenarios will be eventually covered with plan caching  enabled also without impacting test execution time.